contracts/paymaster/src/paymaster.rs

```diff
@@ -0,0 +1,39 @@
+#![no_std]
+
+multiversx_sc::imports!();
+
+pub mod forward_call;
+const FEE_PAYMENT: usize = 0;
+
+/// An empty contract. To be used as a template when starting a new contract from scratch.
+#[multiversx_sc::contract]
+pub trait PaymasterContract: forward_call::ForwardCall {
+    #[init]
+    fn init(&self) {}
+
+    #[endpoint(forwardExecution)]
+    #[payable("*")]
+    fn forward_execution(
+        &self,
+        relayer_addr: ManagedAddress,
+        dest: ManagedAddress,
+        endpoint_name: ManagedBuffer,
+        endpoint_args: MultiValueEncoded<ManagedBuffer>,
+    ) {
+        let payments = self.call_value().all_esdt_transfers();
+        require!(!payments.is_empty(), "There is no fee for payment!");
+
+        let fee_payment = payments.get(FEE_PAYMENT);
+        self.send().direct_esdt(
+            &relayer_addr,
+            &fee_payment.token_identifier,
+            0,
+            &fee_payment.amount,
+        );
```

Comment:
Shouldn't you check that the fee is equal to some amount?

Reply:
that is checked at the relayer level.